src/SimulationControl/SpheralController.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, integrator, kernel,
                  restartBaseName = "restart",
                  restartObjects = [],
                  restartFileConstructor = SiloFileIO,
+                 restartFileCount = 0,
```

Comment:
What does restartFileCount == 0 mean in general?  Do we want this to default to mpi.procs?

Reply:
I set it in SidreFileIO.hh. I figured that it should be set/changed by the class that uses it. Also could make the controller less confusing when using other FileIO, but could be changed if needed.

Reply:
I set it in SidreFileIO.hh. I figured that it should be set/changed by the class that uses it. Also could make the controller less confusing when using other FileIO, but could be changed if needed.

Reply:
I think I have two suggestions here:
- the name "restartFileCount" implies (to me) the total number of restart files in a directory, not per timeslice.  Maybe change the parameter name to something like "numRestartFilesPerTimeslice" or something similarly horribly long and descriptive.  At least in the SpheralController.
- I'd go ahead and make the default mpi.procs as well, since I think that's less confusing.  We can then screen for setting it to something else, and have the SpheralController issue a warning if someone is setting it to something other than mpi.procs and not using SidreFileIO.

Reply:
I think renaming this to something like `SPIOFileCount` might be more descriptive as this is only applicable to when we want to tune SPIO. Defaulting this to `None` might also make things clearer. We can then default to always using the `(filename, access)` constructor even with Sidre. Then using `(filename, access, count)` when `SPIOFileCount` is not `None` and we are using Sidre). This would help us avoid duplicating the default semantics  both in the C++ interface and the python one.

Reply:
I was doing my review as you left your last comment @jmikeowen so I didnt' see it, changing the name to either or a combo of the two if fine with me :)